### PR TITLE
Improved Handshaking process

### LINKED ISSUE #225: Handshake proposal without token collision check
First comment from Olivier's review in https://mailarchive.ietf.org/arch/msg/tsvwg/T-9P8iVoU5i89VK9QJXnOG_IsL4/

```
MPDCCP uses subflows like MPTCP and HMAC to authenticate addresses.

My first concern is the utilization of the Token in MPDCCP which is 
derived from the keys exchanges during the handshake. This design was 
required in MPTCP because there was no space in the extended SYN header 
to carry a connection identifier. In MPTCP, this forces the server to 
check for hash collissions when generating keys. There is no need to 
included this complexity in MPDCCP. MPDCCP would be clearer and probably 
easier to implement with a connection identifier that is selected by the 
host that receives MP_JOIN requests and could be exchanged durign the 
second phase of the handshake for example.
```

### PR DESCRIPTION
**Pull request by markusa**
Closes #225 and #248

### CODE REVIEW

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 1]
```diff
@@ -1270,18 +1268,18 @@ Address A1    Address A2                              Address B1
...
-     |--------- MP_KEY(Key-A) + MP_KEY(Key-B) ------------>|
+     |---------------------------------------------------->|
      |<----------------------------------------------------|
      |   DCCP-Ack  |                                       |
      |             |                                       |
      |             |DCCP-Request + Change R(MP_CAPABLE,...)|
-     |             |--- MP_JOIN(TB,RA) ------------------->|
-     |             |<------MP_JOIN(TB,RB) + MP_HMAC(B)-----|
+     |             |--- MP_JOIN(CI-B,RA) ----------------->|
+     |             |<------MP_JOIN(CI-B,RB) + MP_HMAC(B)---|
```
[the remark this answers is not in the document]
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1282]: OK, I think I tend to agree. 

Perhaps it makes sense to use the CI-A to check again on Host-A whether the relationship CI-A<->CI-B is still correct on Host-B. I will update the draft accordingly.

[the remark this answers is not in the document]
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd]: Updated in https://github.com/tsvwg/ietf-multipath-dccp/pull/269/commits/580bcbaf7683e3fe033ead887967405ace853637. In the Fallback section the case if a wrong CI is sent during MP_JOIN is already handled.

**Comment by markusa** [comment]
@abrunstrom: I changed the text according to our discussion to have a consistent usage of the CI acronym and replaced at one laction `DCCP streams` with `DCCP subflows`.
It can be reviewed now.

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 2]
```diff
@@ -1294,17 +1292,20 @@ Address A1    Address A2                              Address B1
 The basic initial handshake for the first subflow is as follows:
 
 * Host A sends a DCCP-Request with the MP-Capable feature Change
-  request and the MP_KEY option with an Host-specific Key-A for
-  each of supported types as described in {{MP_KEY}}.
+  request and the MP_KEY option with an Host-specific CI-A and a Key-A for
+  each of supported types as described in {{MP_KEY}}. CI-A is a unique identifier during the
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1296]:
```suggestion
  each of the supported key types as described in {{MP_KEY}}. CI-A is a unique identifier during the
```

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 3]
```diff
@@ -1294,17 +1292,20 @@ Address A1    Address A2                              Address B1
...
 * Host B sends a DCCP-Response with Confirm feature for
-  MP-Capable and the MP_Key option with a single Host-specific Key-B.
+  MP-Capable and the MP_Key option with a unique Host-specific CI-B and a single Host-specific Key-B.
   The type of the key is chosen from the list of supported types
   from the previous request.
 
-* Host A sends a DCCP-Ack with both Keys echoed to Host B.
+* Host A sends a DCCP-Ack to confirm the proper key exchange.
+
+* Host B sends a DCCP-Ack to complete the handshaking and set both connection ends to the OPEN state.
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1306]:
```suggestion
* Host B sends a DCCP-Ack to complete the handshake and set both connection ends to the OPEN state.
```

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 4]
```diff
@@ -1313,14 +1314,13 @@ The handshake for subsequent subflows based on a successful initial
 handshake is as follows:
 
 * Host A sends a DCCP-Request with the MP-Capable feature Change
-  request and the MP_JOIN option with Host B’s Token TB, generated from
-  the derived key by applying a SHA256 hash and truncating to the first
-  32 bits. Additionally, an own random nonce RA is transmitted with the
-  MP_JOIN.
+  request and the MP_JOIN option with Host B’s CI-B,
+  taken from the basic initial handshake. Additionally, an own random nonce
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1318]:
```suggestion
  obtained during the initial handshake. Additionally, an own random nonce
```

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 5]
```diff
@@ -1568,11 +1568,11 @@ algorithm documented in Sections {{MP_KEY}} and {{handshaking}}. The
 security of the MP-DCCP connection depends on the use of keys that are
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1568]: Is it still a hash-based handshake algorithm? Not sure if it referred to the generation of the token? Or it refers to the HMAC perhaps?

File: draft-ietf-tsvwg-multipath-dccp.mkd [hunk 6]
```diff
@@ -1568,11 +1568,11 @@ algorithm documented in Sections {{MP_KEY}} and {{handshaking}}. The
...
 shared once at the start of the first subflow and are never sent again
 over the network. To ease demultiplexing while not revealing
-cryptographic material, future subflows use a truncated cryptographic
-hash of this key as the connection identification "token". The keys are
+cryptographic material, future subflows use the initially exchanged
+CI information. The keys exchanged once at the beginning are
 concatenated and used as keys for creating Hash-based Message
 Authentication Codes (HMACs) used on subflow setup, in order to verify
-that the parties in the handshake are the same as in the original
+that the parties in the handshake of future subflows are the same as in the original
```
Comment (abrunstrom) [inline draft-ietf-tsvwg-multipath-dccp.mkd:1575]: "future subflows" sound a bit strange to me here. "of the new subflow" or "of additional subflows"?

**Review by abrunstrom — APPROVED** [review]
Marked a few nits, otherwise looks good I think.

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 6 above)
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd]: changed to "subsequent subflows" in https://github.com/tsvwg/ietf-multipath-dccp/pull/269/commits/3e95e62dd8402ba0021c267fd4020f4a7f04dfd1

File: draft-ietf-tsvwg-multipath-dccp.mkd (hunk 5 above)
Comment (markusa) [reply draft-ietf-tsvwg-multipath-dccp.mkd:1568]: Added reference to MP_HMAC section in https://github.com/tsvwg/ietf-multipath-dccp/pull/269/commits/24f9bcba139584ce3eef3dd9f628c81a7b4de758

**Pull request merged**